Source/Core/DolphinWX/Debugger/MemoryWindow.cpp

```diff
@@ -370,30 +384,33 @@ void CMemoryWindow::OnSearch(wxCommandEvent& event)
     return;
   }
 
-  u8* end = &ram_ptr[ram_size - search_bytes.size() + 1];
-  u8* ptr = &ram_ptr[addr];
-  while (true)
+  u8* ptr;
```

Comment:
Ideally, this if-else should likely be made its own separate function to keep the searching logic fully separate from anything related to the GUI as well as the preceding sanitizing code. I'm also fairly sure `ptr`, `begin` and `end` can all be pointers to const data, as `std::search` and `find_end` don't modify their sequences.